### ISSUE TITLE
ForestDB Views Issue :Router: Router unable to route request to do_GET_DesignDocument(java.lang.NullPointerException)

### ISSUE THREAD

**Issue by NitzDKoder** [couchbase/couchbase-lite-java-core#1535]
Hi Folks,

Project Setup:
1)Cordova Based CBL 1.3.1 with forestDB is been used.
https://github.com/couchbaselabs/Couchbase-Lite-PhoneGap-Plugin

Success case:

Views from Javascript level get data as below.

```
[12/21 12:45:44.941] [2] View: ... Finished re-indexing (getMeta/getMeta)
[12/21 12:45:44.943] [2] View: Query getMeta/getMeta: Found row with key=s919988996002_ptx1482234358303_meta.doc, value={_id=s919988996002_ptx1482234358303_meta.doc, _rev=2-9f32e383e4aea492e94361e145b6dd5c, _revisions={ids=[9f32e383e4aea492e94361e145b6dd5c, 2664b6c6f556bf523553005a83549f5d], start=2}, channels=[PTX-919988996003-meta-CH, PTX-919988996002-meta-CH], errorDescription=, expiration=1482407201, from={name=S5 White 1, senderId=stel:+919988996002}, msgCreateTime=1482234358291, msgPriority=0, msgProperty=8, msgReceivedTime=1482234401, msgText=1, opaqueData=NA, ptxType=[text], recipientType=contact, state=processed, status=0, to=[{recipientId=stel:+919988996003, recipientName=S6-PTX-6003}], type=ptxmeta, ver=1.0, _local_seq=7}, id=s919988996002_ptx1482234358303_meta.doc
[12/21 12:45:44.944] [2] View: Query getMeta/getMeta: Found row with key=s919988996002_ptx1482236314942_meta.doc, value={_id=s919988996002_ptx1482236314942_meta.doc, _rev=2-e04f28ac2431f7f1dba76b7038f1e83a, _revisions={ids=[e04f28ac2431f7f1dba76b7038f1e83a, e84123f661caf0153bd175f6a63648b4], start=2}, channels=[PTX-919988996003-meta-CH, PTX-919988996002-meta-CH], errorDescription=, expiration=1482409157, from={name=S5 White 1, senderId=stel:+919988996002}, msgCreateTime=1482236314924, msgPriority=0, msgProperty=8, msgReceivedTime=1482236357, msgText=101, opaqueData=NA, ptxType=[text], recipientType=contact, state=processed, status=0, to=[{recipientId=stel:+919988996003, recipientName=S6-PTX-6003}], type=ptxmeta, ver=1.0, _local_seq=15}, id=s919988996002_ptx1482236314942_meta.doc
[12/21 12:45:44.944] [2] View: Query getMeta/getMeta: Found row with key=s919988996003_ptx1482235434066_meta.doc, value={_id=s919988996003_ptx1482235434066_meta.doc, _rev=2-10cede0247768b997fe771664617f325, _revisions={ids=[10cede0247768b997fe771664617f325, d900d2a0e6664aa756944f8943d017ed], start=2}, channels=[PTX-919988996002-meta-CH, PTX-919988996003-meta-CH], errorDescription=, expiration=1482408279, from={name=S6-PTX-6003, senderId=stel:+919988996003}, msgCreateTime=1482235434056, msgPriority=0, msgProperty=8, msgReceivedTime=1482235479, msgText=Hello, opaqueData=NA, ptxType=[text], recipientType=contact, state=processed, status=0, to=[{recipientId=stel:+919988996002, recipientName=S5 White 1}], type=ptxmeta, ver=1.0, _local_seq=10}, id=s919988996003_ptx1482235434066_meta.doc
[12/21 12:45:44.945] [2] View: Query getMeta/getMeta: Found row with key=s919988996003_ptx1482236090178_meta.doc, value={_id=s919988996003_ptx1482236090178_meta.doc, _rev=2-03b6e62358748e8ecc382c896aa478c1, _revisions={ids=[03b6e62358748e8ecc382c896aa478c1, ea3e51ee0eb9b194fd902302e5c477d7], start=2}, channels=[PTX-919988996002-meta-CH, PTX-919988996003-meta-CH], errorDescription=, expiration=1482408934, from={name=S6-PTX-6003, senderId=stel:+919988996003}, msgCreateTime=1482236090171, msgPriority=0, msgProperty=8, msgReceivedTime=1482236134, msgText=90, opaqueData=NA, ptxType=[text], recipientType=contact, state=processed, status=0, to=[{recipientId=stel:+919988996002, recipientName=S5 White 1}], type=ptxmeta, ver=1.0, _local_seq=12}, id=s919988996003_ptx1482236090178_meta.doc
[12/21 12:45:44.946] [2] View: Query getMeta/getMeta: Found row with key=s919988996003_ptx1482236295939_meta.doc, value={_id=s919988996003_ptx1482236295939_meta.doc, _rev=2-bd0629c4d11d93bba4b3a2c14906204f, _revisions={ids=[bd0629c4d11d93bba4b3a2c14906204f, 08421af1f57b14f24886971a25ae4165], start=2}, channels=[PTX-919988996002-meta-CH, PTX-919988996003-meta-CH], errorDescription=, expiration=1482409140, from={name=S6-PTX-6003, senderId=stel:+919988996003}, msgCreateTime=1482236295929, msgPriority=0, msgProperty=8, msgReceivedTime=1482236340, msgText=100, opaqueData=NA, ptxType=[text], recipientType=contact, state=processed, status=0, to=[{recipientId=stel:+919988996002, recipientName=S5 White 1}], type=ptxmeta, ver=1.0, _local_seq=14}, id=s919988996003_ptx1482236295939_meta.doc
[12/21 12:45:44.946] [2] View: Query getMeta/getMeta: Found row with key=s919988996003_ptx1482236443607_meta.doc, value={_id=s919988996003_ptx1482236443607_meta.doc, _rev=2-39897a087a9407ddcfed804613884811, _revisions={ids=[39897a087a9407ddcfed804613884811, 718d834dfb78d6a92f14c336e3ab418e], start=2}, channels=[PTX-919988997001-meta-CH, PTX-919988997002-meta-CH, PTX-919988996002-meta-CH, PTX-919988996003-meta-CH], errorDescription=, expiration=1482409288, from={name=S6-PTX-6003, senderId=stel:+919988996003}, msgCreateTime=1482236443600, msgPriority=0, msgProperty=8, msgReceivedTime=1482236488, msgText=1234567, opaqueData=NA, ptxType=[text], recipientType=cggroup, state=processed, status=0, to=[{recipientId=cgtel:+919988996003;kn-corp-groups=1000000001_656, recipientName=STD1}], type=ptxmeta, ver=1.0, _local_seq=18}, id=s919988996003_ptx1482236443607_meta.doc
[12/21 12:45:44.947] [2] View: Query getMeta/getMeta: Found row with key=s919988996003_ptx1482236484870_meta.doc, value={_id=s919988996003_ptx1482236484870_meta.doc, _rev=2-b0cb6bf5da7af335eb16123ee43947c5, _revisions={ids=[b0cb6bf5da7af335eb16123ee43947c5, fbf9b6226e7fb43930b6334fc9790c95], start=2}, channels=[PTX-919988997001-meta-CH, PTX-919988997002-meta-CH, PTX-919988996002-meta-CH, PTX-919988996003-meta-CH], errorDescription=, expiration=1482409329, from={name=S6-PTX-6003, senderId=stel:+919988996003}, msgCreateTime=1482236484858, msgPriority=0, msgProperty=8, msgReceivedTime=1482236529, msgText=Upxoydoyd, opaqueData=NA, ptxType=[text], recipientType=cggroup, state=processed, status=0, to=[{recipientId=cgtel:+919988996003;kn-corp-groups=1000000001_656, recipientName=STD1}], type=ptxmeta, ver=1.0, _local_seq=20}, id=s919988996003_ptx1482236484870_meta.doc
[12/21 12:45:44.948] [2] Listener: RouterCallbackBlock.onResponseReady() START
[12/21 12:45:44.948] [2] Listener: RouterCallbackBlock.onResponseReady() END

```
Failure Case:
Sometimes the Views fails to get data with below error.

```
[12/21 02:35:32.384] [2] View: ... Finished re-indexing (getMeta/getMeta)
[12/21 02:35:32.385] [2] Database: ForestDB Warning: getDocument(docID, true) docID=[s919988996002_ptx1482234358303_meta.doc] error=[com.couchbase.cbforest.ForestException: key not found]
[12/21 02:35:32.386] [2] Router: Router unable to route request to do_GET_DesignDocument
java.lang.reflect.InvocationTargetException
	at java.lang.reflect.Method.invoke(Native Method)
	at com.couchbase.lite.router.Router.start(SourceFile:556)
	at com.couchbase.lite.listener.LiteServlet.service(SourceFile:141)
	at obfuscated.ayt.service(SourceFile:802)
	at Acme.Serve.Serve$ServeConnection.runServlet(SourceFile:2251)
	at Acme.Serve.Serve$ServeConnection.parseRequest(SourceFile:2176)
	at Acme.Serve.Serve$ServeConnection.run(SourceFile:1988)
	at obfuscated.f.run(SourceFile:1223)
	at java.lang.Thread.run(Thread.java:818)
Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'long com.couchbase.lite.internal.RevisionInternal.getSequence()' on a null object reference
	at com.couchbase.lite.store.ForestDBViewStore.regularQuery(SourceFile:414)
	at com.couchbase.lite.View.query(SourceFile:358)
	at com.couchbase.lite.router.Router.queryDesignDoc(SourceFile:2516)
	at com.couchbase.lite.router.Router.do_GET_DesignDocument(SourceFile:2536)
	... 9 more
```

https://github.com/couchbase/couchbase-lite-java-core/blob/1.3.1/src/main/java/com/couchbase/lite/View.java#L358
https://github.com/couchbaselabs/couchbase-lite-java-forestdb/blob/1.3.1/src/main/java/com/couchbase/lite/store/ForestDBViewStore.java#L414

Basically API failed to get DOC revision from forestDB.."key not found" should not come as document is in the DB.
 
Any fix available for NullPointerException..

Pls confirm...

Thanks
Nithin

**Comment by hideki** [comment]
In case of failing to load the linked document, print warning and not set sequence.
https://github.com/couchbase/couchbase-lite-ios/blob/66133e2ec76851ed16828f3b05476de0671ced3e/Source/CBL_ForestDBQueryEnumerator.mm#L191-L195

**Comment by hideki** [comment]
PR: https://github.com/couchbaselabs/couchbase-lite-java-forestdb/pull/91
